### PR TITLE
mdbook builds

### PR DESCRIPTION
**Pull request by saraswatpuneet**
Problem
=======
- [x] Remove remark validate links package dependency and config
- [x] Runs remark
- [x] Runs spellcheck
- [x] Runs lint
- [x] Build mdbook (should also run link-check plugin)
- [x] Build should fail if any of the above fail
Link to GitHub Issue(s): https://github.com/LibertyDSNP/spec/issues/156

Solution
========
- [x] Update package
- [x] Update github workflow to add the steps towards build
- [x]  Add deployment action but comment it out for different PR

Change summary:
---------------
- [x] Updated package
- [x] Renamed ci.yml to mdbook_cicd.yml

Steps to Verify:
----------------
1. Github actions mdbook_cicd should pass

### CODE REVIEW

File: .github/workflows/mdbook_cicd.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Build dsnp spec book 📚
+        run: mdbook build
+
+      - name: Create mdbook build artifact 📦
+        uses: actions/upload-artifact@v3
+        with:
+          name: spec_mdbook
+          path: |
+            book
+
```
Comment (saraswatpuneet) [inline .github/workflows/mdbook_cicd.yml:52]: Complete CI for MDBook,
Deployment to github pages needs to be set i  new PR

File: .github/workflows/mdbook_cicd.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Use Node.js 🐰
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.17.3'
+
+      - name: Update npm to v7 🔼
+        run: npm i -g npm
```
Comment (wilwade) [inline .github/workflows/mdbook_cicd.yml:26]: Our package.json specifies `"engines": {"node": ">=16.14.0 <17"},`. I think we can move this to Node 16. Then we can remove the npm upgrade action. (16 defaults to npm v7+)

File: .github/workflows/mdbook_cicd.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
```
Comment (wilwade) [inline .github/workflows/mdbook_cicd.yml:11]: Might be worth upgrading to v3.0.0 that was just released

File: .github/workflows/mdbook_cicd.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+      - uses: actions/cache@v2
```
Comment (wilwade) [inline .github/workflows/mdbook_cicd.yml:13]: Looks like this one also updated to v3. Might be worth it.

File: .github/workflows/mdbook_cicd.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          node-version: '14.17.3'
+
+      - name: Update npm to v7 🔼
+        run: npm i -g npm
+
+      - name: Install 💾
+        run: npm install
+
+      - name: Lint 🧶
+        run: npm run lint
```
Comment (wilwade) [inline .github/workflows/mdbook_cicd.yml:32]: This runs both lint-md and spellcheck.

**Review by wilwade — CHANGES_REQUESTED** [review]
Nice. Just a few small updates possible and bumping to node 16 for the workflow.

File: package-lock.json [hunk 6]
```diff
@@ -13,8 +13,7 @@
                 "remark": "^14.0.2",
                 "remark-cli": "^10.0.1",
                 "remark-lint": "^9.1.1",
-                "remark-preset-lint-recommended": "^6.1.2",
-                "remark-validate-links": "github:wilwade/remark-validate-links#7a52742ba4d94c088929f4c1ee7fe633d5a03e07"
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply package-lock.json:17]: I think they were previously removed from package.json could not find this dependency

File: .github/workflows/mdbook_cicd.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        run: npm run spellcheck
+
+      - name: Setup mdbook and build dsnp spec book 📚
+        uses: drmingdrmer/mdbook-full@main
+        
+      - name: Build dsnp spec book 📚
+        run: mdbook build
+
+      - name: Create mdbook build artifact 📦
+        uses: actions/upload-artifact@v3
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply .github/workflows/mdbook_cicd.yml:47]: Good point just did

File: .github/workflows/mdbook_cicd.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Use Node.js 🐰
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16.14.2
+
```
Comment (saraswatpuneet) [inline .github/workflows/mdbook_cicd.yml:24]: picking node lts 16.14.2

File: .github/workflows/mdbook_cicd.yml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Install 💾
+        run: npm install
+
+      - name: Lint and Spellcheck 🧶
+        run: npm run lint
+
```
Comment (saraswatpuneet) [inline .github/workflows/mdbook_cicd.yml:30]: npm run lint runs both md lint, spelling checking

File: .github/workflows/mdbook_cicd.yml (hunk 1 above)
Comment (saraswatpuneet) [reply .github/workflows/mdbook_cicd.yml:45]: Tag the assets and archive

File: .github/workflows/mdbook_cicd.yml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          name: spec_mdbook
+          path: spec_mdbook.tar
+
+      #- name: Deploy mdBook to github pages
+      #   uses: peaceiris/actions-gh-pages@v3
+      #   if: ${{ github.ref == 'refs/heads/main' }}
+      #   with:
+      #     github_token: ${{ secrets.GITHUB_TOKEN }}
+      #     publish_dir: ./book
```
Comment (saraswatpuneet) [inline .github/workflows/mdbook_cicd.yml:51]: Enabled in PR https://github.com/LibertyDSNP/spec/pull/164

**Review by wilwade — APPROVED** [review]
👍

**Pull request merged**